internal/config/main.go

```diff
@@ -52,82 +66,58 @@ type LocalCert struct {
 	Content []byte
 }
 
-var (
-	parameter = FortiExporterParameter{
-		AuthFile:      flag.String("auth-file", "fortigate-key.yaml", "file containing the authentication map to use when connecting to a Fortigate device"),
+func setParameterFlags() *FortiExporterParameter {
```

Comment:
From the other comment you said you modify global flag state. I assumed the goal of this code here was to remove depdendency on global flag state, if it's not - why move the flag initializers into this function instead of using the pretty common pattern of defining them in the file's `var` as we did before?

Reply:
The target of this refactoring is to remove the global config state, which allows to drop this `reinit` logic altogether and ease the use of custom config values during testing.     

This is used for testing so that all flag pointers can be be recreated.     
What I referencing with global flag and args is the attributes of the imported packages `flag.CommandLine` and `os.Args`     

It hasn't been in scope for this PR to use Flagsets instead of flag, but I will look into it.